### PR TITLE
Initial checkin of speed performance benchmarking script.

### PR DESCRIPTION
**Pull request by alexaryn**

### CODE REVIEW

File: examples/bench.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    .partition(partitioner=SycamorePartitioner(extract_table_structure=True, extract_images=True))
+    .regex_replace(COALESCE_WHITESPACE)
+    .extract_entity(entity_extractor=OpenAIEntityExtractor("title", llm=davinci_llm, prompt_template=title_template))
+    .mark_bbox_preset(tokenizer=tokenizer)
+    .merge(merger=MarkedMerger())
+    .spread_properties(["path", "title"])
+    .split_elements(tokenizer=tokenizer, max_tokens=512)
+    .explode()
+    .sketch()
+    .embed(embedder=SentenceTransformerEmbedder(model_name="sentence-transformers/all-MiniLM-L6-v2", batch_size=100))
```
Comment (bsowell) [inline examples/bench.py:46]: Did we want to switch this to gte-small?

File: examples/bench.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+davinci_llm = OpenAI(OpenAIModels.GPT_3_5_TURBO_INSTRUCT.value)
+tokenizer = HuggingFaceTokenizer("thenlper/gte-small")
+
+ctx = sycamore.init()
+
+ds = (
+    ctx.read.binary(paths, binary_format="pdf", filesystem=fsys)
+    .partition(partitioner=SycamorePartitioner(extract_table_structure=True, extract_images=True))
+    .regex_replace(COALESCE_WHITESPACE)
+    .extract_entity(entity_extractor=OpenAIEntityExtractor("title", llm=davinci_llm, prompt_template=title_template))
```
Comment (bsowell) [inline examples/bench.py:39]: I'm sorry I haven't sent you the extract_properties call. I just need to find a couple of minutes to fix it, but who knows when I will get it. I can add it as a separate PR, I guess.

File: examples/bench.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+# code may also need to be changed to adjust actor pools.  The
+# script should be run similar to this:
+#
+# TIMETRACE=/tmp/tt poetry run python examples/bench.py
+
+import sys
+import pyarrow.fs
+
+# ruff: noqa: E402
+sys.path.append("../sycamore")
```
Comment (bsowell) [inline examples/bench.py:13]: Is this necessary?

File: examples/bench.py (hunk 1 above)
Comment (alexaryn) [reply examples/bench.py:48]: Yes, let's.

File: examples/bench.py (hunk 3 above)
Comment (alexaryn) [reply examples/bench.py:13]: Seems like since the monorepo reorg, no.

**Review by bsowell — APPROVED** [review]

**Comment by bsowell** [comment]
Oh, I guess fix the linting if you can, but otherwise LGTM.

**Pull request merged**